pkg/apis/okctl.io/v1alpha1/cluster_v1alpha1.go

```diff
@@ -25,14 +27,14 @@ type Cluster struct {
 	// Metadata uniquely identifies a cluster.
 	Metadata ClusterMeta `json:"metadata"`
 
-	// Github defines what organisation, repository, team, etc. that
+	// Github defines what organisation, repository, etc. that
 	// this cluster will integrate with.
 	Github ClusterGithub `json:"github"`
 
-	// PrimaryDNSZone defines the main primary zone to associate with this
-	// cluster. This will be the zone that we will use to create domains
+	// ClusterRootURL defines the main primary zone to associate with this
+	// cluster. This will be the zone that we will use to create subdomains
 	// for auth, ArgoCD, etc.
-	PrimaryDNSZone ClusterDNSZone `json:"primaryDNSZone"`
+	ClusterRootURL string `json:"clusterRootURL"`
```

Comment:
A [url](https://en.wikipedia.org/wiki/URL) is a very specific thing, so this is misleading. `PrimaryDNSZone` isn't much better though.. 🤔 

Maybe something like:

`BaseDomain`: also slightly misleading, since it strictly only includes domain and TLD
`SubDomain`: probably the most accurate, but maybe not what we want to communicate
`RootDomain`: kind of combines some of these

In reality, we don't really require it to be a subdomain, if someone wants to create their own domain, they are free to do so.

Maybe we should just call it:

`DefaultClusterDomain`: the default domain used for all kinds of grafana setups and stuff. Also leaves room for multiple domains in the future if someone would like to do so

I don't know, but I don't think we should have `URL` in there at leaast

Reply:
Agree, URL is incorrect. Domain makes sense. I'll vote for `RootDomain` for its simplicity. `DefaultClusterDomain` could work too, but I think it raises more questions than being explanatory. But maybe with a good comment it can work.

Reply:
We went for `ClusterRootDomain`, as it was hopefully more descriptive than just `RootDomain`.